src/flows/context/query.tsx

```diff
@@ -255,12 +255,20 @@ export const QueryProvider: FC = ({children}) => {
         return raw
       })
       .then(raw => {
-        return {
-          source: text,
-          raw: raw.csv,
-          parsed: fromFlux(raw.csv),
-          error: null,
-        } as FluxResult
+        return new Promise((resolve, reject) => {
+          requestAnimationFrame(() => {
```

Comment:
this was an attempt to offload some of the freezing that was happening on the page while it was parsing 4 csv files at once. it helped a little, but i think moving parsing into a service worker is the next step